test/components/componentTests.ts

```diff
@@ -246,333 +344,338 @@ describe("Component behavior", () => {
-    c.renderTo(svg);
-    let gs = (<any> c)._element.selectAll("g");
-    let g0 = d3.select(gs[0][0]);
-    let g1 = d3.select(gs[0][1]);
-    let g2 = d3.select(gs[0][2]);
-    let g3 = d3.select(gs[0][3]);
-    assert.isTrue(g0.classed("background-container"), "the first g is a background container");
-    assert.isTrue(g1.classed("content"), "the second g is a content container");
-    assert.isTrue(g2.classed("foreground-container"), "the third g is a foreground container");
-    assert.isTrue(g3.classed("box-container"), "the fourth g is a box container");
-    svg.remove();
-  });
+    describe("on fixed size components", () => {
+      let fixedWidth = 100;
+      let fixedHeight = 100;
 
-  it("component defaults are as expected", () => {
-    assert.strictEqual(c.xAlignment(), "left", "x alignment defaults to \"left\"");
-    assert.strictEqual(c.yAlignment(), "top", "y alignment defaults to \"top\"");
-    let layout = c.requestedSpace(1, 1);
-    assert.strictEqual(layout.minWidth, 0, "requested minWidth defaults to 0");
-    assert.strictEqual(layout.minHeight, 0, "requested minHeight defaults to 0");
-    svg.remove();
-  });
+      beforeEach(() => {
+        TestMethods.fixComponentSize(c, fixedWidth, fixedHeight);
+        c.anchor(svg);
+      });
 
-  it("fixed-width component will align to the right spot", () => {
-    TestMethods.fixComponentSize(c, 100, 100);
-    c.anchor(svg);
-    c.xAlignment("left").yAlignment("top");
-    c.computeLayout();
-    assertComponentXY(c, 0, 0, "top-left component aligns correctly");
+      it("gives the component the fixed space if offered more space", () => {
+        c.computeLayout();
+        assert.strictEqual(c.width(), fixedWidth, "width bounded to fixed width");
+        assert.strictEqual(c.height(), fixedHeight, "height bounded to fixed height");
+        c.destroy();
+        svg.remove();
+      });
 
-    c.xAlignment("center").yAlignment("center");
-    c.computeLayout();
-    assertComponentXY(c, 150, 100, "center component aligns correctly");
+      it("gives the component the offered space if it is less than the fixed space", () => {
+        let restrictedWidth = fixedWidth / 2;
+        let restrictedHeight = fixedHeight / 2;
+        c.computeLayout({x: 0, y: 0}, restrictedWidth, restrictedHeight);
+        assert.strictEqual(c.width(), restrictedWidth, "width bounded to restricted width");
+        assert.strictEqual(c.height(), restrictedHeight, "height bounded to restricted height");
+        c.destroy();
+        svg.remove();
+      });
 
-    c.xAlignment("right").yAlignment("bottom");
-    c.computeLayout();
-    assertComponentXY(c, 300, 200, "bottom-right component aligns correctly");
-    svg.remove();
-  });
+      it("does not translate if more space was requested than offered", () => {
+        let requestedWidth = SVG_WIDTH * 2;
+        let requestedHeight = SVG_HEIGHT * 2;
+        let c = new Plottable.Component();
```

Comment:
c has been defined before
also, `c = makeFixedSizeComponent(width, height)` is more readable